library/src/com/google/maps/android/geojson/GeoJsonLayer.java

```diff
@@ -0,0 +1,270 @@
+
+    /**
+     * Gets an iterable of all GeoJsonFeature elements that have been added to the layer
+     *
+     * @return iterable of GeoJsonFeature elements
+     */
+    public Iterable<GeoJsonFeature> getFeatures() {
+        return mRenderer.getFeatures();
+    }
+
+    /**
+     * Adds all the GeoJsonFeature objects parsed from the given GeoJSON data onto the map. Default
+     * styles are applied if the features haven't been previously added to the layer or if the
+     * layer has been cleared.
+     */
+    public void addLayer() {
+        for (GeoJsonFeature feature : mRenderer.getFeatures()) {
+            addFeature(feature);
+        }
+    }
+
+    /**
+     * Adds a GeoJsonFeature to the layer. If the point, linestring or poylgon style is set to
+     * null, the relevant default styles are applied.
+     *
+     * @param feature GeoJsonFeature to add to the layer
+     */
+    public void addFeature(GeoJsonFeature feature) {
+        // Check if no style and apply default styles
+        if (feature.getPointStyle() == null) {
+            feature.setPointStyle(mDefaultPointStyle);
+        }
+        if (feature.getLineStringStyle() == null) {
+            feature.setLineStringStyle(mDefaultLineStringStyle);
+        }
+        if (feature.getPolygonStyle() == null) {
+            feature.setPolygonStyle(mDefaultPolygonStyle);
+        }
+        mRenderer.addFeature(feature);
+    }
+
+    /**
+     * Removes the given GeoJsonFeature from the layer
+     *
+     * @param feature feature to remove
+     */
+    public void removeFeature(GeoJsonFeature feature) {
+        mRenderer.removeFeature(feature);
+    }
+
+    /**
+     * Gets the map on which the layer is rendered
+     *
+     * @return map on which the layer is rendered
+     */
+    public GoogleMap getMap() {
+        return mRenderer.getMap();
+    }
+
+    /**
+     * Renders the layer on the given map. The layer on the current map is removed and
+     * added to the given map.
+     *
+     * @param map to render the layer on, if null the layer is cleared from the current map
+     */
+    public void setMap(GoogleMap map) {
+        mRenderer.setMap(map);
+    }
+
+    /**
+     * Clears all of the GeoJsonFeatures from the layer. All styles are removed from the
+     * GeoJsonFeatures.
+     */
+    public void clearLayer() {
+        mRenderer.removeLayerFromMap();
+    }
+
+    /**
+     * Gets the default style used to render GeoJsonPoints
+     *
+     * @return default style used to render GeoJsonPoints
+     */
+    public GeoJsonPointStyle getDefaultPointStyle() {
+        return mDefaultPointStyle;
+    }
+
+    /**
+     * Sets the default style to use when rendering GeoJsonPoints. This style is only applied to
+     * GeoJsonPoints that are added after this method is called.
+     *
+     * @param geoJsonPointStyle to set as the default style for GeoJsonPoints
+     */
+    public void setDefaultPointStyle(GeoJsonPointStyle geoJsonPointStyle) {
```

Comment:
this default doesn't appear to be used for the geojson specified in the file passed in.